### PR TITLE
Adds 2d x32 binary and unary elementwise microkernels.

### PR DESCRIPTION
**Pull request by stellaraccident**
I got the list of ops to start with by adding a new option to LowerLinalgMicrokernels to warn on slow-path ops and running it over the TOSA test suite. The uniform-type binary ops work all the way through. The unary ops are stubbed out for the moment. I have not yet handled comparison and ternary ops (select), which compose most of the rest of the standard fare. There are substantial tensor-level optimizations needed after that to get more things onto the fast path, but this is a start.

This patch does several things:

* Renames the mmt4d builtin to "ukernel" and adds support for elementwise. I retained some optional control in the vmvx/module.c so things can be pruned later. I suspect this is the level we want to be doing that at anyway.
* Decided to shard the elementwise ops by rank and type width, implementing the 2d and x32 variants. The actual primitives will be different for each width, so that seemed a reasonable cut point.
* Extends LowerLinalgMicrokernels to lower all supported x32 binary and unary opcodes.
* Fixes more fallout from more complicated buffered programs going through the lowering flow:
  * Added a descriptor resolver for allocations
  * Added a `get_raw_interface_binding_buffer` to handle the case where we were splitting early to work on buffers vs memrefs and needing to get at the raw buffer (not after the lowering, which introduces offsets, etc). I'm not sure this is the best way to do this (or that there is a best way).

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir [hunk 1]
```diff
@@ -47,6 +46,315 @@ vm.import @add.2d.f32(
...
+
+  %sizes : tuple<i64, i64>
+)
+
+//===----------------------------------------------------------------------===//
+// VMVX Unary Elementwise Kernels
+// Each is specialized by opcode, rank and type width.
+//===----------------------------------------------------------------------===//
+
+vm.import @absf.2d.x32(
```
Comment (benvanik) [inline compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir:278]: should these be `abs.2d.f32`, given that they only operate on f32? (they can all use the same shim internally, but it's kind of confusing to have both a typed op name and generic untyped suffix)

Comment (stellaraccident) [reply compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir:278]: Ah, I was just relying too much on some automated bits. Yes, I can change them to typed.

Comment (benvanik) [reply compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir]: they still have the redundant naming in them (vmvx.abs**f**.2d.**f**32) - I think you need to change the enum?

Comment (stellaraccident) [reply compiler/src/iree/compiler/Dialect/VMVX/vmvx.imports.mlir]: Was debating that, but yeah... Ok, I can remove the 'f' and 'i' but still need to keep 's' and 'u' on some of them.

**Review by benvanik — APPROVED** [review]

**Pull request merged**